File: src/main/java/net/lomeli/trophyslots/core/network/MessageOpenWhitelist.java
package net.lomeli.trophyslots.core.network;

import io.netty.buffer.ByteBuf;

import cpw.mods.fml.common.network.simpleimpl.IMessage;
import cpw.mods.fml.common.network.simpleimpl.IMessageHandler;
import cpw.mods.fml.common.network.simpleimpl.MessageContext;

import net.lomeli.trophyslots.TrophySlots;

public class MessageOpenWhitelist implements IMessage, IMessageHandler<MessageOpenWhitelist, IMessage> {

    public MessageOpenWhitelist() {
    }

    @Override
    public void fromBytes(ByteBuf buf) {
    }

    @Override
    public void toBytes(ByteBuf buf) {
    }

    @Override
    public IMessage onMessage(MessageOpenWhitelist message, MessageContext ctx) {
        TrophySlots.proxy.openWhitelistGui();
        return null;
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/network/MessageOpenWhitelist.kt
package net.lomeli.trophyslots.core.network

import io.netty.buffer.ByteBuf

import net.lomeli.trophyslots.TrophySlots
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext

public class MessageOpenWhitelist() : IMessage, IMessageHandler<MessageOpenWhitelist, IMessage> {

    override fun toBytes(buf: ByteBuf?) {
    }

    override fun fromBytes(buf: ByteBuf?) {
    }

    override fun onMessage(message: MessageOpenWhitelist?, ctx: MessageContext?): IMessage? {
        TrophySlots.proxy?.openWhitelistGui()
        return message
    }
}